### PR TITLE
STREAMLINE-722: Switch kafka source and sink to storm-kafka-client

### PR DESCRIPTION
**Pull request by priyank5485**
Change-Id: I35bda085a86fa39483a9836cca164788ace9ee34

### CODE REVIEW

**Comment by priyank5485** [comment]
WIP -- Do not merge. Still need to test this. Also need to add jaas config property to communicate securely.

**Comment by HeartSaVioR** [comment]
@priyank5485 
If you need to get Kafka principal, you can get it from `kafka_principal_name` for Service Configuration `kafka-env` in Service `KAFKA`.

File: streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroKafkaSpoutTranslator.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AvroKafkaSpoutTranslator implements RecordTranslator<Object, ByteBuffer> {
```
Comment (satishd) [inline streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroKafkaSpoutTranslator.java]: Most part of the code in this class is from `AvroKafkaSpoutScheme`. So, it is better to rename that class to `AvroKafkaSpoutTranslator`  and have all the required modifications including renaming that class and implementing the appropriate interfaces etc, as it is a healthy practice to keep the history of the changes without dropping them whenever it is possible.

File: streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/KafkaSpoutFluxComponent.java [hunk 2]
```diff
@@ -86,178 +50,177 @@ private String addSpoutConfigComponent () {
...
-        validateStringFields(optionalStringFields, false);
+    private String addRecordTranslator () {
+        String translatorId = "avroKafkaSpoutTranslator" + UUID_FOR_COMPONENTS;
+        String translatorClassname = "com.hortonworks.streamline.streams.runtime.storm.spout.AvroKafkaSpoutTranslator";
+        List<Object> constructorArgs = new ArrayList<>();
+        constructorArgs.addAll(getConstructorArgsYaml(new String[]{TopologyLayoutConstants.JSON_KEY_OUTPUT_STREAM_ID, TopologyLayoutConstants
+                .JSON_KEY_TOPIC}));
+        constructorArgs.add(kafkaSource != null ? kafkaSource.getId() : "");
+        constructorArgs.add(conf.get(TopologyLayoutConstants.SCHEMA_REGISTRY_URL));
+        constructorArgs.add(conf.get("readerSchemaVersion") != null ? Integer.parseInt((String) conf.get("readerSchemaVersion")) : null);
```
Comment (satishd) [inline streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/KafkaSpoutFluxComponent.java]: @harshach found out this issue
agree that readerSchemaVersion integer value should be null for empty and null strings.

**Comment by harshach** [comment]
@satishd in the interest of time, I merged this into master. We are less than 4 weeks away for a GA Release and need to get the builds to QE and tests running. 
A lot to catch up let's figure out these things after the release.

**Pull request merged**